packages/block-library/src/site-title/edit.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * WordPress dependencies
+ */
+import { useEntityProp, useEntitySaving } from '@wordpress/core-data';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { RichText } from '@wordpress/block-editor';
+
+export default function SiteTitleEdit() {
+	const [ title, setTitle ] = useEntityProp( 'root', 'site', 'title' );
+	const [ isDirty, isSaving, save ] = useEntitySaving( 'root', 'site', 'title' );
```

Comment:
I'm not sure this hook is going to be very useful. I think it's necessary now because we have this temporary UI but we'll only be needed in a single place, in the end, making the hook an unnecessary API.

Reply:
It will be very useful for any plugin author creating a UI that persists to an entity. Think beyond blocks or even the editor.

Reply:
I think plugin authors will probably first look for actions in the core-data store as it's a more common pattern for performing actions. I'm also not sure we should be optimizing for plugin authors at this stage, I'd prefer extensibility APIs to be based on feedback and use-cases.

Reply:
If the goal of entities is to abstract away the transient and persistent editing of data records and we have a hook for getting and setting transiently, it feels very natural to also have a hook for setting persistently with some helpful indicators of whether there is anything to save and whether a save is already ongoing.

Think about how much more complex this component would be without it. I also think that we will inevitably have blocks that need local saving. This could even be used for reusable blocks as well.

We could make it experimental for now?

Reply:
> We could make it experimental for now?

This would work, I think we need to be very careful in general when adding APIs. We should also take a look from time to time at old experimental APIs and make decisions after some time but that's a different topic.

Reply:
Done!

Yeah, we could scrape the codebase to plot experimental APIs vs the time they were created at and work our way from older to newer.